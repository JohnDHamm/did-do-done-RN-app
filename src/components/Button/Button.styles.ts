import styled from 'styled-components/native';
import { FONTS } from '../../styles';

export const Container = styled.View<{ color: string; border: string }>`
  padding: 3px 36px;
  align-items: center;
  background-color: ${(props) => props.color};
  border-color: ${(props) => props.border};
  border-radius: 5px;
  border-width: 2px;
`;

export const Label = styled.Text<{ labelColor: string }>`
  font-family: ${FONTS.PRIMARY};
  font-size: 22px;
  color: ${(props) => props.labelColor};
`;
