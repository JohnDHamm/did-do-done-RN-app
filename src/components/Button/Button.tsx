import React from 'react';
import { Container, Label } from './Button.styles';
import { COLORS } from '../../styles';

interface ButtonProps {
  label: string;
  type?: 'regular' | 'alt' | 'delete';
}

const Button: React.FC<ButtonProps> = ({
  label,
  type = 'regular',
}: ButtonProps) => {
  let backgroundColor = COLORS.PRIMARY_PURPLE;
  let borderColor = COLORS.PRIMARY_PURPLE;
  let labelColor = COLORS.WHITE;

  switch (type) {
    case 'alt':
      backgroundColor = COLORS.WHITE;
      labelColor = COLORS.PRIMARY_PURPLE;
      break;
    case 'delete':
      backgroundColor = COLORS.WHITE;
      borderColor = COLORS.MISSED_RED;
      labelColor = COLORS.MISSED_RED;
      break;
  }
  return (
    <Container color={backgroundColor} border={borderColor}>
      <Label labelColor={labelColor}>{label}</Label>
    </Container>
  );
};

export default Button;
