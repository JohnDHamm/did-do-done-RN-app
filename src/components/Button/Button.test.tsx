import React from 'react';
import { cleanup, render } from 'react-native-testing-library';
import Button from './Button';

describe('Button', () => {
  afterEach(cleanup);

  it('renders correctly with defaults', () => {
    const testRenderer = render(<Button label="regular" />).toJSON();

    expect(testRenderer).toMatchSnapshot();
  });

  it('renders correctly as an "alt" type', () => {
    const testRenderer = render(<Button label="alt" type="alt" />).toJSON();

    expect(testRenderer).toMatchSnapshot();
  });

  it('renders correctly as an "delete" type', () => {
    const testRenderer = render(
      <Button label="delete" type="delete" />
    ).toJSON();

    expect(testRenderer).toMatchSnapshot();
  });

  it('renders the label prop', () => {
    const { getByText } = render(<Button label="Button 1" />);

    expect(getByText('Button 1')).not.toBeNull();
  });
});
