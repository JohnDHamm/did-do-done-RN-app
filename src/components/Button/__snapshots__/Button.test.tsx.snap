// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly as an "alt" type 1`] = `
<View
  border="#4c408f"
  color="transparent"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderColor": "#4c408f",
        "borderRadius": 5,
        "borderWidth": 2,
        "paddingBottom": 3,
        "paddingLeft": 36,
        "paddingRight": 36,
        "paddingTop": 3,
      },
    ]
  }
>
  <Text
    labelColor="#4c408f"
    style={
      Array [
        Object {
          "color": "#4c408f",
          "fontFamily": "Lobster-Regular",
          "fontSize": 22,
        },
      ]
    }
  >
    alt
  </Text>
</View>
`;

exports[`Button renders correctly as an "delete" type 1`] = `
<View
  border="#B70000"
  color="#B70000"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#B70000",
        "borderColor": "#B70000",
        "borderRadius": 5,
        "borderWidth": 2,
        "paddingBottom": 3,
        "paddingLeft": 36,
        "paddingRight": 36,
        "paddingTop": 3,
      },
    ]
  }
>
  <Text
    labelColor="#fff"
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Lobster-Regular",
          "fontSize": 22,
        },
      ]
    }
  >
    delete
  </Text>
</View>
`;

exports[`Button renders correctly with defaults 1`] = `
<View
  border="#4c408f"
  color="#4c408f"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#4c408f",
        "borderColor": "#4c408f",
        "borderRadius": 5,
        "borderWidth": 2,
        "paddingBottom": 3,
        "paddingLeft": 36,
        "paddingRight": 36,
        "paddingTop": 3,
      },
    ]
  }
>
  <Text
    labelColor="#fff"
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Lobster-Regular",
          "fontSize": 22,
        },
      ]
    }
  >
    regular
  </Text>
</View>
`;
